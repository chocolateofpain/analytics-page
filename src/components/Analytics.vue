<template>
  <div>
    <Typography as="h2">
      Reporting
    </Typography>
    <div>Analyse long term performance of your checks and export aggregated metrics for external reporting duties.</div>
      <CheckItem v-for="(item, index) in checks" :key='index' v-bind:checkItemData="item" />
  </div>
</template>

<script>
import '@primer/octicons/build/build.css'
import axios from 'axios';
import CheckItem from './CheckItem.vue'
import Typography from './library/typography/Typography.vue'

export default {
  name: 'Analytics',
  components: {
    CheckItem,
    Typography
  },
  data () {
    return {
      checks: null
    }
  },
  created () {
    axios
      .get('http://localhost:3000/checks')
      .then(response => (this.checks = response.data))
  }
}
</script>
