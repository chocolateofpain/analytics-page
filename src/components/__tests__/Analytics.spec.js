import { mount } from '@vue/test-utils'
import Analytics from '../Analytics.vue'

describe('Analytics', () => {
  it('should be defined', () => {
    expect(Analytics).toBeDefined()
  })

  describe('snapshots', () => {
    it('should render component', () => {
      const component = mount(Analytics)
      expect(component).toMatchSnapshot()
    })
  })
})
