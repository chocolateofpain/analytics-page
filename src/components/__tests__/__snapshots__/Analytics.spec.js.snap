// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics snapshots should render component 1`] = `
<div class="analytics-container align-left">
  <div class="analytics-content">
    <p class="analytics-title typography typography--p      ">
      Reporting
    </p> <small class="typography typography--small      ">
      Analyse long term performance of your checks and export aggregated metrics for external reporting duties.
    </small>
    <table class="checks-table">
      <colgroup>
        <col class="type-col">
        <col class="name-col">
        <col span="4" class="data-cols">
      </colgroup>
      <thead class="row-height table-header">
        <th class="type-col">type</th>
        <th class="name-col">name</th>
        <th class="data-col">success</th>
        <th class="data-col">avg</th>
        <th class="data-col">p95</th>
        <th class="data-col">p99</th>
      </thead>
      <tbody>
        <td colspan="6"></td>
      </tbody>
    </table>
  </div>
</div>
`;
